src/objectivec/test/ort_genai_api_test.mm

```diff
@@ -0,0 +1,138 @@
+
+    OGAGeneratorParams *params = [[OGAGeneratorParams alloc] initWithModel:model error:&error];
+    ORTAssertNullableResultSuccessful(params, error);
+
+    OGATensor* tensor = [[OGATensor alloc] initWithDataPointer:data.data() shape:shape type:OGAElementTypeFloat32 error:&error];
+    ORTAssertNullableResultSuccessful(tensor, error);
+
+    ret = [params setModelInput:@"test_input" tensor:tensor error:&error];
+    ORTAssertBoolResultSuccessful(ret, error);
+}
+
+- (void)testGetOutput {
+    std::vector<int64_t> input_ids_shape{2, 4};
+    NSArray<NSNumber*>* input_ids = @[@0, @0, @0, @52, @0, @0, @195, @731];
+    const auto batch_size = input_ids_shape[0];
+    int max_length = 10;
+
+    NSError *error = nil;
+    BOOL ret = NO;
+    OGAModel* model = [[OGAModel alloc] initWithPath:[ORTGenAIAPITest getModelPath] error:&error];
+    ORTAssertNullableResultSuccessful(model, error);
+
+    OGAGeneratorParams *params = [[OGAGeneratorParams alloc] initWithModel:model error:&error];
+    ORTAssertNullableResultSuccessful(params, error);
+
+    [params setSearchOption:@"max_length" doubleValue:max_length error:&error];
+    XCTAssertNil(error);
+
+    [params setSearchOption:@"batch_size" doubleValue:batch_size error:&error];
+    XCTAssertNil(error);
+
+    OGAGenerator* generator = [[OGAGenerator alloc] initWithModel:model
+                                                           params:params
+                                                            error:&error];
+    ORTAssertNullableResultSuccessful(generator, error);
+    [generator appendTokens:input_ids error:&error];
+    XCTAssertNil(error);
+
+    // check prompt
+     // full logits has shape [2, 4, 1000]. Sample 1 for every 200 tokens and the expected sampled logits has shape [2, 4, 5]
+    std::vector<float> expected_sampled_logits_prompt{0.29694548f, 0.00955007f, 0.0430819f, 0.10063869f, 0.0437237f,
+                                                      0.27329233f, 0.00841076f, -0.1060291f, 0.11328877f, 0.13369876f,
+                                                      0.30323744f, 0.0545997f, 0.03894716f, 0.11702324f, 0.0410665f,
+                                                      -0.12675379f, -0.04443946f, 0.14492269f, 0.03021223f, -0.03212897f,
+                                                      0.29694548f, 0.00955007f, 0.0430819f, 0.10063869f, 0.0437237f,
+                                                      0.27329233f, 0.00841076f, -0.1060291f, 0.11328877f, 0.13369876f,
+                                                      -0.04699047f, 0.17915794f, 0.20838135f, 0.10888482f, -0.00277808f,
+                                                      0.2938929f, -0.10538938f, -0.00226692f, 0.12050669f, -0.10622668f};
+
+    OGATensor* prompt_logits_ptr = [generator getOutput:@"logits" error:&error];
+    ORTAssertNullableResultSuccessful(prompt_logits_ptr, error);
+    auto prompt_logits = static_cast<float*>([prompt_logits_ptr getDataPointerWithError:&error]);
```

Comment:
also check `error` here

Reply:
These two assertions are tricky because `ORTAssertNullableResultSuccessful` expects an `id`, instead of native C++ pointer. Using it will cause error:

```text
Cannot initialize a variable of type '__strong id' with an lvalue of type 'float *'
```

Do you want me to add another one for native C++ pointer? Or maybe just use `ORTAssertNullableResultSuccessful` and checks for `prompt_logits != nullptr`?

Reply:
oh, `XCTAssertNotNil` expects `id`.

how about `XCTAssertNil(error)` and `XCTAssertNotEqual(prompt_logits, nullptr)`? up to you if you want to add another helper macro.

Reply:
Using `XCTAssertNil(error)` and `XCTAssertNotEqual(prompt_logits, nullptr)` seems to be working fine.